### PR TITLE
Add EOS on chain event parsing (3/4)

### PR DESCRIPTION
**Pull request by gitmp01**
#### Notes:

 - `PAM.isAuthorized()` ~ 1.8M gas

### CODE REVIEW

File: solidity/src/contracts/PAM.sol [hunk 1]
```diff
@@ -163,28 +184,42 @@ contract PAM is Ownable, IPAM {
         return true;
     }
 
+    function _fromHexCharToUint8(uint8 x) internal pure returns (uint8) {
```
Comment (envin3) [inline solidity/src/contracts/PAM.sol:187]: nice

File: solidity/src/contracts/PAM.sol [hunk 2]
```diff
@@ -110,7 +109,28 @@ contract PAM is Ownable, IPAM {
 
         offset += 32 * 3; // skip other topics
 
-        if (!this.doesContentMatchOperation(eventPayload[offset:], operation))
+        // Checking the protocol id against 0x02 (EOS chains)
```
Comment (envin3) [inline solidity/src/contracts/PAM.sol:112]: 🫡

File: javascript/event-attestator/src/ProofcastEventAttestator.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ class ProofcastEventAttestator {
     return concat([
       zeroPadValue(Buffer.from(event.account, 'utf-8'), 32),
       ...topics,
-      event.data,
+      Buffer.from(JSON.stringify(event.data), 'utf-8'),
```
Comment (envin3) [inline javascript/event-attestator/src/ProofcastEventAttestator.js:58]: stringify needs to operate on data encoded with JSON.parse, fixed on https://github.com/pnetwork-association/pnetwork-v4/pull/71

**Review by envin3 — COMMENTED** [review]
LGTM

**Comment by gitmp01** [comment]
Better late than never 😅, sorry for not letting you know, I'll ping you privately next time

**Pull request merged**